### DISCUSSION TITLE
[問題] 用python做parse

### DISCUSSION THREAD

**Discussion by LeoTsai999** [1101-GS4719-C/Python-Community-GS4719-C#3383] [Q&A]
這是在ptt上看到的一篇關於python的小問題，覺得很有趣，想和大家分享一下。

之前沒學過python

google到的教學文章有範例 但卻不懂原理..

請問一下我今天假使要parse證交所的收盤價
http://0rz.tw/n8qLj

是要先自己分析網頁的原始碼 找出我要的資料在哪裡
然後再parse嗎?

原始碼長得像這樣子http://ppt.cc/IhfX

請問該如何做parse

麻煩給點指導 謝謝！

--

當你要解析網頁上的資料，比如證交所的收盤價時，的確需要先分析網頁的原始碼，找到你想要提取的資料位置。以下是一些步驟和建議來幫助你理解這個過程：

1. **分析網頁原始碼**：
   - 打開網址後，使用瀏覽器的「檢查元素」功能來觀察網頁的 HTML 結構。尋找包含你想要的收盤價的標籤，比如 `table`、`tr` 和 `td` 等。
   - 確定資料的層級，找到正確的標籤位置，這將幫助你在程式中定位要提取的資料。

2. **使用適合的工具進行解析**：
   - Python 提供了幾個強大的網頁解析庫，例如 `BeautifulSoup` 和 `lxml`。這些庫可以幫助你輕鬆解析 HTML 內容並提取所需資訊。
   - 建議從 `requests` 套件開始，這是一個用來發送 HTTP 請求的庫，配合 `BeautifulSoup` 或 `lxml` 來解析回應的 HTML。

3. **範例流程**：
   - 利用 `requests` 獲取網頁資料。
   - 使用 `BeautifulSoup` 來解析獲得的 HTML 內容，根據前面分析的結果提取特定的元素。
   - 最後，將提取的數據儲存、顯示或進行後續分析。

4. **創新性建議**：
   - 理解資料的來源和使用權限很重要，確保遵守資料使用的規範。在進行網頁爬蟲時，注意網站的 `robots.txt` 檔案，查看可否合法爬取該網站的數據。
   - 可以考慮使用 `pandas` 套件將提取的資料轉換成資料框（DataFrame），這樣可以簡化數據處理和分析的過程。

5. **學習方向**：
   - 可以考慮學習 Python 的基本語法、資料結構，後續再深入爬蟲及數據分析領域，這些能力將使你在資料科學和分析領域有更大的發揮空間。
   - 參加一些線上的課程（如 Coursera 或 Udemy）來獲得更多系統化的知識，在學習過程中可以逐步實踐相關專案，增強技能。

希望這些建議能幫助你順利解析證交所的收盤價！如果有其他問題，歡迎隨時詢問。

**Comment by laveeshb** [comment]
## Feature Parity — Data Source

The [Product Releases Tracker](https://github.com/orgs/github/projects/2463/views/259) (project #2463) is the source of truth for feature gaps. From the first 100 items:

| Status | Count |
|--------|-------|
| Fully shipped | 74 |
| **Shipped (pending GHES)** | **23** |
| Planned | 2 |
| Paused | 1 |

3,794 total items in the tracker. The **Shipped (pending GHES)** items are features fully available on dotcom but not yet on GHES/Proxima — these are the feature parity gaps to inventory and prioritize.

Key fields to analyze: `Release Phase`, `Product Pillar`, `GHES version`, `Target Quarter`.
